app/models/tax_household.rb

```diff
@@ -43,9 +43,22 @@ def group_by_year
   end
 
   def eligibile_csr_kind(family_member_ids)
+    # send shopping family members ids as input
     thh_members = tax_household_members.where(:applicant_id.in => family_member_ids)
-    thh_m_eligibility_kind = thh_members&.pluck(:csr_eligibility_kind)&.uniq
-    (thh_members.pluck(:is_ia_eligible).include?(false) || thh_m_eligibility_kind.count == 0) ? 'csr_0' : eligibile_csr_kind_for_shopping(thh_m_eligibility_kind)
+    thhm_appid_csr_hash = thh_members.inject({}) do |result, thhm|
+      result[thhm.applicant_id] = thhm.csr_eligibility_kind
+      result
+    end
+
+    if FinancialAssistanceRegistry.feature_enabled?(:native_american_csr)
```

Comment:
Checks if the feature is enabled to consider AI/AN status for CSR determination.